src/arrayMethodSort.js

```diff
@@ -5,8 +5,23 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const haveArgs = arguments.length > 0;
+
+    for (let iterator = this.length - 1; iterator >= 0; iterator--) {
+      for (let i = 0; i < iterator; i++) {
+        const current = this[i];
+        const next = this[i + 1];
+
+        if ((haveArgs && compareFunction(current, next) > 0)
+        || (!haveArgs && `${current}` > `${next}`)) {
```

Comment:
it's not really readable and not easy to understand what's done in this line
that's where a `defaultCompareFunction` would be very useful in particular